tests/integration/relations/test_backend_database.py

```diff
@@ -105,6 +107,41 @@ async def test_pgbouncer_stable_when_deleting_postgres(ops_test: OpsTest):
             ),
         )
 
-        await ops_test.model.applications[PG].remove()
-        wait_for_relation_removed_between(ops_test, PG, PGB)
-        await ops_test.model.wait_for_idle(apps=[PGB], status="active", timeout=1000)
+        await scale_application(ops_test, PGB, 3)
+        await asyncio.gather(
+            ops_test.model.wait_for_idle(
+                apps=[PGB], status="active", timeout=1000, wait_for_exact_units=3
+            ),
+            ops_test.model.wait_for_idle(
+                apps=[PG], status="active", timeout=1000, wait_for_exact_units=3
+            ),
+        )
+
+        username = f"relation_id_{relation.id}"
```

Comment:
A reminder for me and you: I need to fix the k8s charm to create the username following [another pattern](https://github.com/canonical/postgresql-operator/pull/15#discussion_r940469355) and it should be changed later here too.